analyzers/src/SonarAnalyzer.Common/Helpers/GeneratedCodeRecognizer.cs

```diff
@@ -55,7 +55,9 @@ public abstract class GeneratedCodeRecognizer
         protected abstract string GetAttributeName(SyntaxNode node);
 
         public bool IsGenerated(SyntaxTree tree) =>
-            HasGeneratedFileName(tree) || HasGeneratedCommentOrAttribute(tree);
+           !string.IsNullOrEmpty(tree.FilePath)
+           && (HasGeneratedFileName(tree) || HasGeneratedCommentOrAttribute(tree))
+           && !IsRazorGeneratedFile(tree);
```

Comment:
Should be slightly more performant in this order:
```suggestion
           !string.IsNullOrEmpty(tree.FilePath)
           && !IsRazorGeneratedFile(tree)
           && (HasGeneratedFileName(tree) || HasGeneratedCommentOrAttribute(tree));
```

Reply:
Why do you think it would?

The reason I put it like this is to be less "intrusive" and more politically correct. We will only check if the file is razor/cshtml generated file once we for sure know it is a generated file. This way we will not change the old behavior for any non generated file. If, I move the check before like you suggested, for every non generated file I will have 4 checks to see if they are any of the razor/cshtml generated files + check if they are generated at all. So, basically there will be 5 endsWith checks instead of 1 for all non generated files. (And in theory there should be more non generated files than generated)

Reply:
You are right, I was thinking from a Razor project analysis point of view which might have more `.razor` files.